include/RAJA/policy/PolicyBase.hpp

```diff
@@ -69,16 +69,26 @@ enum class Pattern {
   taskgraph,
 };
 
+enum class Platform {
+  undefined = 0,
+  host = 1,
+  cuda = 2,
+  omp_target = 4
+};
+
 struct PolicyBase {
 };
 
 template <Policy P = Policy::undefined,
           Launch L = Launch::undefined,
-          Pattern Pat = Pattern::undefined>
+          Pattern Pat = Pattern::undefined,
+          Platform Plat = Platform::host
```

Comment:
I'd look at what you do with omp_.*_exec, though. It's a tough thing to work with. Also, don't know if OMP4 goes through omp_.*_exec or is its own thing, but might get interesting there. If you've looked at the tradeoffs and decided tying policies to execution spaces is fine, that's totally kosher, I just want to make sure the tradeoff is a conscious decision